### PR TITLE
test: eliminate leaked-thread test pollution + guard against it

### PR DESCRIPTION
**Pull request by eous**
## Problem

Background daemons, event loops, and test servers that outlived their test bled into *later* tests' captured output — surfacing as intermittent `ValueError: I/O operation on closed file` (a daemon logging into pytest's already-closed capture). It's a heisenbug (only reproduces across the full suite, never a single module) and the same class of pollution behind a past multi-day CI-hang investigation.

A diagnostic pass found the real leakers: the collector's `console-discovery` thread, docker_healthcheck's HTTP test servers, and the MCP pool-auth tests' background event loops (`asyncio_N`) + FastMCP uvicorn upstreams.

## What this does

**Guard (so the next one is caught in minutes, not days)**
- `tests/conftest.py`: `_no_leaked_threads` — an autouse fail-on-leak guard that snapshots live threads at setup and **fails any test that leaves a thread running past teardown** (5s shared grace-join), with an `@pytest.mark.allow_thread_leak` opt-out.
- `logging.raiseExceptions = False` — mutes the benign logging-handler-vs-capture-teardown race (a pytest artifact; production stderr isn't closed mid-run).
- Shared teardown helpers `stop_loop_thread` / `serve_until_exit`.

**Product fix**
- `turnstone/console/collector.py`: the node-discovery loop slept uninterruptibly (`time.sleep(discovery_interval)`), so `ClusterCollector.stop()` couldn't join the `console-discovery` thread until the full interval elapsed — **a real shutdown hang in production** (up to `discovery_interval`, default 60s). It now sleeps on an interruptible `Event` that `stop()` sets and `start()` clears.

**Test-fixture teardown fixes**
- docker_healthcheck: `_serve` → a `serve` fixture-factory that shuts down its HTTP servers + joins their threads.
- MCP pool-auth (5 files): `running_loop_mgr` loops close cleanly (`shutdown_default_executor` + `close`); FastMCP uvicorn upstreams stop via `timeout_graceful_shutdown=0` + `force_exit` (graceful shutdown was hanging on a held-open streamable-http stream).

## Testing
- `ruff` + `mypy` clean; reviewed via `/review` (1 minor + nits, all applied; the minor was the collector `start()` wake-clear).
- **Full non-live suite: 7456 passed, 0 "I/O operation on closed file", 0 leaked-thread failures** — and ~1.5 min faster (2:53 vs 4:41), since the leaked servers/loops were dragging it.

## Follow-up (separate)
Production uvicorn (server.py / console/server.py) doesn't bound `timeout_graceful_shutdown`. It likely shuts down fine (sse_starlette closes SSE streams on the exit signal, unlike FastMCP's transport here), but bounding it would be cheap defensive hardening against a SIGTERM-hang edge case — out of scope for this branch.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,17 +1,115 @@
...
+    main = threading.main_thread()
+    current = threading.current_thread()
+    # One deadline shared across all joined threads — a deliberate TOTAL
+    # teardown budget (not per-thread), so a pathological test can't stall
+    # teardown by N×grace. A genuine never-stopping leak exhausts it and fails.
+    deadline = time.monotonic() + _THREAD_LEAK_GRACE
+    leaked = []
+    for t in threading.enumerate():
+        if t.ident in before or t is main or t is current or not t.is_alive():
+            continue
```
[the remark this answers is not in the document]
Comment (eous) [reply tests/conftest.py]: Fixed in c58bb062. Good catch — `Thread.ident` is recycled after a thread exits, so an ident-based snapshot could mistake a new leaked thread (reusing an exited thread's ident) for a pre-existing one and miss the leak (false negative). Switched to snapshotting the `Thread` objects (`before = set(threading.enumerate())`) and comparing by object identity. Full non-live suite re-verified green (7456 passed, 0 leaked-thread failures).

File: tests/test_docker_healthcheck.py [hunk 2]
```diff
@@ -52,13 +52,31 @@ def log_message(self, *args):
...
-    threading.Thread(target=httpd.serve_forever, daemon=True).start()
-    return httpd.server_address[1]
+@pytest.fixture
+def serve():
+    """Factory that starts an HTTP(S) server on an ephemeral port and returns it.
+
+    Every server it starts is shut down + its serve_forever thread joined at
+    teardown, so the thread never outlives the test (which would otherwise bleed
+    into a later test's captured output / leak the listener).
+    """
```
[the remark this answers is not in the document]
Comment (eous) [reply tests/test_docker_healthcheck.py]: Fixed in c58bb062 — reworded the docstring to "returns that port" (the factory returns the ephemeral port number, not the server object).
